### PR TITLE
Port stethoscopes + innate verbs from nyano

### PR DESCRIPTION
**Pull request by Elijahrane**
## About the PR 
Adds stethoscopes, which display different messages depending on what the target's total asphyxiation damage is.

Also adds innate verbs, a new verb type similar to utility verbs, but supplied by components on the user themselves.

**Changelog**

:cl: Rane
- add: Stethoscopes have been ported over from Nyanotrasen. Wear one on your neck and right click on someone else to listen to their chest.

### CODE REVIEW

**Comment by mirrorcult** [comment]
with stethoscopes we need to think about removing handheld health analyzers at some point

**Comment by mirrorcult** [comment]
or heavily gating them behind research

**Review by mirrorcult — CHANGES_REQUESTED** [review]
so the only issue i really have here is that i dont think `InnateVerb` should exist. verbs are inherently not really supposed to be an innate thing--our closest equivalent to 'innate' stuff are actions, which would be much cleaner for this. we even have item actions, which would let you remove the 'add stethoscope wearer component on equip' stuff. so IMO the stethoscope stuff should just be an entity target item action granted by wearing a stethoscope. this way you can reap all the advantages of actions like support for cooldowns, interaction highlighting, etc

**Comment by Elijahrane** [comment]
I really don't see it as a massive departure from utility verbs, it's functionally one of those without the restriction of the source needing to be in your active hand.

Actions are starting to get to the point where we're going to go above 10 in fairly normal gameplay conditions and have a bunch of people very confused they suddenly need to learn how to use multiple pages

**Comment by mirrorcult** [comment]
need second opinion, maybe @ElectroJr

**Review by ElectroJr — CHANGES_REQUESTED** [review]
If innate verbs should be added, they should actually be innate to the user (e.g., species abilities), and not due to an item they are wearing. Instead of giving the user a `WearingStethoscopeComponent`, utility verbs should just be extended so that they can get verbs from either held or worn items. I.e., also relay the get-verbs event to any equipped clothing, presumably with information about what slot or hand the item is in, so that stethoscopes don't work inside of pockets.

**Comment by ElectroJr** [comment]
As to the general question of whether some interaction should have a verb, action, or both, this question isn't really specific to innate verbs. E.g., the ability to un-anchor something using a wrench is currently a verb, but you could also make it so that holding a wrench gives someone a entity-target unwrench action. The spill-solution verb on beakers could also be an action.

As far as verbs are concerned, unless it ends up making the verb menu too cluttered, I'm of the opinion that any interaction a user can perform involving some specific entity should have a corresponding verb in that entity's verb menu. So either way I think the stethoscope interaction should have a verb. The one exception is maybe innate actions without any target (i.e., scream, toggle combat-mode), though even those could maybe be added to the verb menu when right clicking your own player, but I'm iffy on that. But I have no objection to innate verbs in general. 

As to what interactions should get actions, I think there should actually be some restrictions, because the toolbar does get cluttered much more quickly than the verb menu. This is partially because it shows all available actions, not just those applicable to a specific entity, and partially because it is just limited to showing 10 actions at a time. I guess much like verbs you could always give everything an action, but with auto-populate set to false so that they don't automatically appear in the action toolbar. That way users could decide on their own to assign those actions to a 0-9, but adding a whole bunch of mostly unused actions probably isn't worth the effort.

So I think a good guideline for whether something should get an (auto-populating) action would be:
- Interactions innate to the player that have no other other dedicated keybinds should all have actions. 
  - E.g., scream should have an action, but not swap hands or drop item. 
  - If there were a dedicated scream button, the action wouldn't be required anymore.
  - Obviously things like spells and misc species abilities will probably never have dedicated key binds, so these should always have actions.
- Actions should be somewhat static. So things like clothing adding actions is probably fine, but actions from hand-held objects are dubious.
  - This maybe wouldn't be so bad if  actions had persistent locations in the toolbar slot. But currently if you have two items giving actions, but then you dropped them and picked them up again, the ordering on the action bar associated and the 0-9 key bindings might change, leading to confusion.
  - On a related note, actions shouldn't be situational. I.e., no stop-drop-and-roll or unbuckle action, leave that to alerts & verbs.
- Any interaction that has a lengthy cooldown or limited charges benefits from having an action, as that gives visual feedback.  So even with a dedicated keybinding, the scream action would still have value.
- Use/Toggle item interactions should get actions only if they are relatively frequently used and if they cannot trivially be triggered some other way.
  - E.g., toggle light for a hardsuit helmet is frequently used, but without the action would require expanding the inventory UI, mousing over the helmet, and then using `E` or the verb menu. Being able to just press `3` or click on the action is more convenient
  - Conversely, something like a hand-held flashlight, which can just be toggled via `z` or `x`+`z` probably shouldn't have an action, it just takes up a toolbar slot. Or at the very least, it should have auto populate set to false. 
- Interact-using interactions should probably not have actions because they can always be trivially triggered by just clicking on the target while holding the item. If you wanted to use an entity-target action you would still have to do that anyways.
  - The one benefit of these actions might would be that you can perform the interaction without having to have the item in your active active hand, but that is pretty minor.
- If an item enables some unclear or unintuitive ability or interaction, it should probably have an action
  - The mere fact that actions will appear as an icon, with a name and description probably help a lot when it comes to discovering the uses/abilities of an item.
  - e.g., now that hard-suits have toggleable helmets, if that feature was ONLY available via a verb, how many users would actually be able to figure it out without using the wiki or asking other players?
  - examine-descriptions can help I guess, but having them explicitly direct people to use the verb menu feels icky.
  - also oops forgot to add a verb for toggleable clothing

Overall I think the stethoscopes should definitely get an action (in addition to the verb), largely because of the last point. I expect that a new user would intuitively try to click on other players while holding one, and not ever think to equip it and then right-click on patients. Giving them an action at least makes it clear that a new interaction/ability was enabled by equipping the item.

**Comment by Elijahrane** [comment]
OK, I'll add an action too and also make it an inventory relay event to cut down as bookkeeping as previously discussed. Using actions for discoverability alone is something that may need to be reevaluated in the future still, I feel.

File: Content.Server/Medical/Stethoscope/StethoscopeSystem.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+        [Dependency] private readonly PopupSystem _popupSystem = default!;
+        [Dependency] private readonly DoAfterSystem _doAfterSystem = default!;
+        [Dependency] private readonly SharedActionsSystem _actionsSystem = default!;
+        public override void Initialize()
+        {
+            base.Initialize();
+            SubscribeLocalEvent<StethoscopeComponent, GotEquippedEvent>(OnEquipped);
+            SubscribeLocalEvent<StethoscopeComponent, GotUnequippedEvent>(OnUnequipped);
+            SubscribeLocalEvent<WearingStethoscopeComponent, GetVerbsEvent<InnateVerb>>(AddStethoscopeVerb);
+            SubscribeLocalEvent<StethoscopeComponent, GetItemActionsEvent>(OnGetActions);
```
Comment (ElectroJr) [inline Content.Server/Medical/Stethoscope/StethoscopeSystem.cs:30]:
```suggestion
            SubscribeLocalEvent<StethoscopeComponent, GetItemActionsEvent>(OnGetActions);
            SubscribeLocalEvent<StethoscopeComponent, StethoscopeActionEvent>(OnStethoscopeAction);
```

File: Content.Server/Medical/Stethoscope/Components/StethoscopeComponent.cs [hunk 2]
```diff
@@ -0,0 +1,28 @@
...
+        public CancellationTokenSource? CancelToken;
+
+        [DataField("delay")]
+        public float Delay = 2.5f;
+
+        public EntityTargetAction Action = new()
+        {
+            Icon = new SpriteSpecifier.Texture(new ResourcePath("Clothing/Neck/Misc/stethoscope.rsi/icon.png")),
+            Name = "stethoscope-verb",
+            Priority = -1,
```
Comment (ElectroJr) [inline Content.Server/Medical/Stethoscope/Components/StethoscopeComponent.cs:24]:
```suggestion
            Priority = -1,
            Whitelist = new() { Components = new string[] { "MobState" } },
```

**Review by ElectroJr — APPROVED** [review]
Other than fixing the action, should be fine to merge

File: Content.Server/Medical/Stethoscope/Components/StethoscopeComponent.cs (hunk 2 above)
Comment (Elijahrane) [reply Content.Server/Medical/Stethoscope/Components/StethoscopeComponent.cs:24]: I'm planning on merging the 2 target actions soonTM and might end up changing that, (it's currently unused anywhere)

**Comment by Elijahrane** [comment]
Review addressed

**Pull request merged**